### ISSUE TITLE
Incorrect non_shorthand_field_patterns warning

### ISSUE THREAD

**Issue by kuviman** [rust-lang/rust#65924]
```rust
struct Foo {
    int: i32,
    text: String,
}

fn foo(foo: &Foo) {
    match *foo {
        Foo { int, text: ref text } => {
            println!("{:?}, {:?}", int, text);
        }
    }
}
```

This code gives incorrect warning:

```
warning: the `text:` in this pattern is redundant
 --> src/lib.rs:8:24
  |
8 |             Foo { int, text: ref text } => {
  |                        -----^^^^^^^^^
  |                        |
  |                        help: remove this
  |
  = note: `#[warn(non_shorthand_field_patterns)]` on by default
```

The suggested fix makes `text` move, which is impossible since `foo` is borrowed.

Using auto borrowing with `match foo` instead of `match *foo` would make all fields borrowed, but I want most of them moved (copied).

[Link to playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=8e648231da2401ffca26633d082c7690)

**Comment by csmoe** [comment]
it's suggesting to remove `text:`, not `ref text` as:
```rust
Foo { int, ref text }
```
